src/components/Layout.jsx

```diff
@@ -15,17 +16,19 @@ const Layout = () => {
   return (
     <BrowserRouter>
       <div className="page-container">
-        <CategoryState>
-          <SearchState>
-            <Navbar />
-            <div className="container">
-              <div className="main">
-                <AppRoutes />
+        <CartState>
+          <CategoryState>
+            <SearchState>
+              <Navbar />
+              <div className="container">
+                <div className="main">
+                  <AppRoutes />
+                </div>
               </div>
-            </div>
-            <Footer />
-          </SearchState>
-        </CategoryState>
+              <Footer />
+            </SearchState>
```

Comment:
This is one of the tradeoffs of using useContext, when using redux, you just need to wrap your components once, just to make you notice on of the main difference!